Update the watchdog timing requirements

fkjagodzinski opened the pull request:
Update the requirements according to https://github.com/ARMmbed/mbed-os/pull/11203, that differentiates lower timeout limit for calibrated and uncalibrated watchdog clock.

This should wait until https://github.com/ARMmbed/mbed-os/pull/11203 is merged to Mbed OS repo.

fkjagodzinski committed:
Update the watchdog timing requirements

Update the requirements according to PR #11203 that differentiates lower
timeout limit for calibrated and uncalibrated watchdog clock.

AnotherButler commented on docs/porting/target/Watchdog.md:
```diff
@@ -11,7 +11,8 @@ Implement the ResetReason API when implementing the Watchdog API. The ResetReaso
 - The function `hal_watchdog_init` is safe to call repeatedly. The function's implementation must not do anything if `hal_watchdog_init` has already initialized the hardware watchdog timer.
 - `UINT32_MAX` milliseconds is the maximum allowed max_timeout `hal_watchdog_get_platform_features()` returns; minimum timeout is 1 ms.
-- The watchdog should trigger at or after the timeout value.
+- The uncalibrated watchdog should trigger at or after the timeout value multiplied by the frequency accuracy ratio of its oscillator (typical_frequency / max_frequency).
```
Does the slash represent division, or does it mean "or" here?

fkjagodzinski replied:
Division; Do you think we should use the word instead of the symbol here?

AnotherButler replied:
Yes, I think that would clarify here. I'll make that change.

Commit:
Clarify slash in Watchdog.md

Clarify slash, as discussed in comments.

AnotherButler reviewed (approved):
Thanks for the PR 👍

AnotherButler commented:
The code dependency has merged to master but has a future tag on it, so I'm waiting to merge it.

fkjagodzinski committed:
Update watchdog timing requirements for deepsleep

Allow the watchdog reset to be delayed up to twice the timeout value in
deepsleep mode according to #11774.

fkjagodzinski commented:
There is another update for the watchdog timing. This time only for the deepsleep mode (based on https://github.com/ARMmbed/mbed-os/issues/11774).

Since this PR is waiting, I pushed changes here. @AnotherButler, I hope this is acceptable; please let me know if this should land on a new docs PR. I'll remove the last commit in that case.

jamesbeyond reviewed (approved):
Looks good

Commit:
Fix spelling in Watchdog.md

Add space for consistent spelling.

fkjagodzinski commented:
https://github.com/ARMmbed/mbed-os/pull/11773 has been merged into master.

iriark01 commented:
Merging PR, as code was merged to master.
